app/components/metadata-panel.tsx

```diff
@@ -406,17 +412,47 @@ const Header = ({ onClose }: { onClose: () => void }) => {
 };
 
 const TabPanelGeneral = ({
-  datasetIri,
   dataSource,
+  chartConfigs,
 }: {
-  datasetIri: string;
   dataSource: DataSource;
+  chartConfigs: ChartConfig[];
 }) => {
   const classes = useOtherStyles();
+  const locale = useLocale();
 
+  const cubes = uniqBy(
+    chartConfigs.flatMap((x) => x.cubes).map((x) => ({ iri: x.iri })),
+    (x) => x.iri
+  );
+
+  const [{ data, fetching, error }] = useDataCubesMetadataQuery({
+    variables: {
+      sourceType: dataSource.type,
+      sourceUrl: dataSource.url,
+      locale,
+      cubeFilters: cubes,
+    },
+  });
+  const cubesMetadata = data?.dataCubesMetadata;
+  if (!cubesMetadata) {
+    return null;
+  }
   return (
     <TabPanel className={classes.tabPanel} value="general">
-      <DataSetMetadata dataSetIri={datasetIri} dataSource={dataSource} />
+      {fetching ? <Loading /> : null}
+      {error ? (
+        <Error>{`${error instanceof Error ? error.message : error}`}</Error>
```

Comment:
So `error` could be something different than `Error`? 🤯

Reply:
Yes, every thing can be throw ;)

```
> const hello = () => { try { throw 'world' } catch (e) { console.log('got thrown', e) } }
undefined
> hello()
got thrown world
```